src/server/qgsserver.cpp

```diff
@@ -346,16 +350,23 @@ void QgsServer::handleRequest( QgsServerRequest &request, QgsServerResponse &res
       printRequestParameters( parameterMap, logLevel );
 
       //Config file path
-      QString configFilePath = configPath( *sConfigFilePath, parameterMap );
-
-      // load the project if needed and not empty
-      const QgsProject *project = mConfigCache->project( configFilePath );
       if ( ! project )
       {
-        throw QgsServerException( QStringLiteral( "Project file error" ) );
-      }
+        QString configFilePath = configPath( *sConfigFilePath, parameterMap );
 
-      sServerInterface->setConfigFilePath( configFilePath );
+        // load the project if needed and not empty
+        project = mConfigCache->project( configFilePath );
+        if ( ! project )
+        {
+          throw QgsServerException( QStringLiteral( "Project file error" ) );
+        }
+
+        sServerInterface->setConfigFilePath( configFilePath );
+      }
+      else
+      {
+        sServerInterface->setConfigFilePath( project->fileName() );
```

Comment:
Do we need this ?

Reply:
@dmarteau not sure if you are talking about `setConfigFilePath`, if this is the case, I think that it is necessary because the `sServerInterface` needs to know the project path. But I think that the method name should be probably renamed to `setProjectPath` etc.

Reply:
btw, what does filename return in this case? I suppose it's an empty string.

Reply:
It can be an empty string, but it depends on the way the project has been build!